api/v1beta1/types.go

```diff
@@ -590,9 +594,6 @@ type SubnetSpec struct {
 	// +optional
 	ID string `json:"id,omitempty"`
 
-	// Name defines a name for the subnet resource.
-	Name string `json:"name"`
-
 	// SecurityGroup defines the NSG (network security group) that should be attached to this subnet.
 	// +optional
 	SecurityGroup SecurityGroup `json:"securityGroup,omitempty"`
```

Comment:
I think we wanted to move ResouceGroup from VnetPeeringSpec to VnetPeeringClassSpec (so it's available in both AzureClusterTemplate & AzureCluster)

(map key should not be added though)

(just had to put the comment somewhere, it's not related to the current line in any way)